drv/update-api/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+
+//use core::cell::Cell;
+use userlib::{sys_send, FromPrimitive};
+use zerocopy::{AsBytes, FromBytes};
+
+#[derive(FromPrimitive)]
```

Comment:
If you add `derive-idol-err` to dependencies, you can do `#[derive(FromPrimitive, IdolError)]` and skip the manual `From` implementations below!